File: 2023/advent_of_code/test/q24/q24_test.exs
defmodule AdventOfCode.Q24Test do
  use ExUnit.Case
  doctest AdventOfCode.Q24

  test "Q24 Part 1 Example works" do
    file_stream = File.stream!("./test/q24/q24_example.txt")
    result = AdventOfCode.Q24.part1(file_stream)
    assert result == 94
  end

  test "Q24 Part 1 works" do
    file_stream = File.stream!("./test/q24/q24.txt")
    result = AdventOfCode.Q24.part1(file_stream)
    assert result == 16939
  end
end

File: 2022/AdventOfCode/src/test/kotlin/q24/Q24Test.kt
package q24

import junit.framework.TestCase.assertEquals
import java.nio.file.Paths
import kotlin.test.Test

class Q24Test {
    @Test
    fun `part 1 ex works`()  {
        val filePath = javaClass.getResource("/q24_ex.txt")?.toURI()?.let { Paths.get(it) }
        assertEquals(18, part1(filePath!!))
    }

    @Test
    fun `part 1 works`()  {
        val filePath = javaClass.getResource("/q24.txt")?.toURI()?.let { Paths.get(it) }
        assertEquals(277, part1(filePath!!))
    }

    @Test
    fun `part 2 ex works`()  {
        val filePath = javaClass.getResource("/q24_ex.txt")?.toURI()?.let { Paths.get(it) }
        assertEquals(54, part2(filePath!!))
    }

    @Test
    fun `part 2 works`()  {
        val filePath = javaClass.getResource("/q24.txt")?.toURI()?.let { Paths.get(it) }
        assertEquals(877, part2(filePath!!))
    }
}